### PR TITLE
Add datatype QC for annotations related to date

### LINKED ISSUE #1800: Change range in annotation related to date to `xsd:dateTime` instead of `xsd:string`
There're 754 cases where the `oboInOwl:creation_date` annotation uses the range `xsd:string`.

- [x] Replace the 754 cases
- [x] Add QC to check the range of annotations related to the date

A sample:
```
AnnotationAssertion(oboInOwl:creation_date obo:CL_0001034 "2012-01-12T09:58:38Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0001035 "2011-11-16T04:28:16Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0001060 "2010-01-06T03:43:27Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002000 "2010-04-26T10:14:16Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002001 "2010-01-13T03:19:22Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002002 "2010-01-13T03:19:28Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002003 "2010-04-26T10:14:30Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002004 "2010-04-26T10:31:50Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002005 "2010-01-13T03:43:10Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002006 "2010-01-13T03:43:12Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002007 "2010-01-14T03:25:15Z"^^xsd:string)
AnnotationAssertion(oboInOwl:creation_date obo:CL_0002008 "2010-01-14T03:25:17Z"^^xsd:string)
```

### PR DESCRIPTION
**Pull request by anitacaron**
Fixes #1800

Need to update the sparql query in the ODK to `xsd:dateTime` instead of `xsd:date`

### CODE REVIEW

**Comment by dosumis** [comment]
Will be fixed centrally in ODK.

File: src/sparql/replace_string_to_datetime.ru [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+INSERT {
+    ?sub ?property ?new_value.
+}
+
+WHERE {
+    VALUES ?property {dct:date dct:issued dct:created oboInOwl:creation_date}
+    ?sub ?property ?value .
+    FILTER(Datatype(?value)!=xsd:dateTime)
+    BIND(STRDT(?value, xsd:dateTime) as ?new_value)
```
Comment (matentzn) [inline src/sparql/replace_string_to_datetime.ru:20]: cool, never new STRDT!

File: src/sparql/illegal-date-violation.sparql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+PREFIX dct: <http://purl.org/dc/terms/>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
+
+SELECT DISTINCT ?term ?property ?value WHERE
+{
+  VALUES ?property {dct:date dct:issued dct:created oboInOwl:creation_date}
+  ?term ?property ?value .
+  FILTER (datatype(?value) != xsd:date || !regex(str(?value), '^\\d{4}-\\d\\d-\\d\\d$'))
```
Comment (matentzn) [inline src/sparql/illegal-date-violation.sparql:9]: I think the logic is slightly wrong, check this:

```suggestion
  FILTER (
    ( 
       datatype(?value) != xsd:date && !regex(str(?value), '^\\d{4}-\\d\\d-\\d\\d$')
     ) || ( 
       datatype(?value) != xsd:dateTime && !regex(str(?value), '^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}Z')
     )
   )
```

Either the value is an xsd:date with the respective format, or it is an xsd:dateTime with the other format.

I could be wrong, and I did not check the syntax..

File: src/ontology/cl-edit.owl
Comment (matentzn) [inline src/ontology/cl-edit.owl:1]: Can you review the diff once yourself? 2649 additions and 2636 removals sounds like there have been some gains..

File: src/sparql/illegal-date-violation.sparql (hunk 2 above)
Comment (anitacaron) [reply src/sparql/illegal-date-violation.sparql:9]: If so, it needs to change in ODK.

Comment (matentzn) [reply src/sparql/illegal-date-violation.sparql:9]: I trust your reading, what do you think?

File: src/ontology/cl-edit.owl
Comment (anitacaron) [reply src/ontology/cl-edit.owl:1]: Plus, I ran `normalize_src` after the update query, and it created some prefixes.

File: src/sparql/illegal-date-violation.sparql (hunk 2 above)
Comment (anitacaron) [reply src/sparql/illegal-date-violation.sparql:9]: I tested your suggestion, and it returns all the correct cases. Having two filters for the same value is a way to have OR.

Comment (matentzn) [reply src/sparql/illegal-date-violation.sparql:9]: > Having two filters for the same value is a way to have OR.

I did not know that.. Completely shocked!

Comment (anitacaron) [reply src/sparql/illegal-date-violation.sparql:9]: It's working as expected, so I'll leave like this.

**Pull request merged**